src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames/bind';
+import styles from './TextInput.scss';
+const cx = classnames.bind( styles );
+
+const TextInput = ( {
+	autoComplete,
+	ariaDescribedBy,
+	disabled,
+	isMultiline,
+	inputRef,
+	id,
+	label,
+	maxLength,
+	placeholder,
+	onBlur,
+	onChange,
+	onClick,
+	onFocus,
+	onInvalid,
+	pattern,
+	readOnly,
+	required,
+	type,
+} ) => {
+	// Class names used by both <input> and <textarea>.
+	const sharedClassNames = {
+		disabled,
+		readOnly,
+		[type]: type,
+	};
+
+	return (
+		<label className={styles.container}>
+			{label && (
+				<span className={`${styles.label} ${required && label ? styles.requiredLabel : ''}`}>
```

Comment:
if you have conditional classnames, just use `cx`

Reply:
for this particular use case, tho, i'd probably do this

```suggestion
				<span className={`${styles.label} ${styles.required}`}>
```

and this change the css to:

```
.label {
  ...
  &.required {
  }
}